File: hw/chalan/u3/hw27GUIsOperation/MODULOS/JAVA/StoreFruit/src/main/java/utils/MongoDBUtil.java
package utils;

/**
 *
 * @author Kevin Chalan, OBJECT MASTER, OOP
 */
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class MongoDBUtil {

    private static final String CONNECTION_STRING =
            "mongodb+srv://kevin:kevin2001@cluster0.oxinj5p.mongodb.net/contac";

    private static final String DATABASE_NAME = "fruit_store";

    private static MongoDatabase database;

    public static MongoDatabase getDatabase() {
        if (database == null) {
            MongoClient client = MongoClients.create(CONNECTION_STRING);
            database = client.getDatabase(DATABASE_NAME);
            createCollectionIfNotExists();
        }
        return database;
    }

    private static void createCollectionIfNotExists() {
        boolean exists = false;

        for (String name : database.listCollectionNames()) {
            if (name.equals("fruits")) {
                exists = true;
                break;
            }
        }

        if (!exists) {
            database.createCollection("fruits");

            database.getCollection("fruits").insertOne(
                    new Document("name", "Apple")
                            .append("price", 1.5)
                            .append("stock", 20)
            );
            database.getCollection("fruits").insertOne(
                    new Document("name", "Banana")
                            .append("price", 0.8)
                            .append("stock", 30)
            );
            database.getCollection("fruits").insertOne(
                    new Document("name", "Orange")
                            .append("price", 1.2)
                            .append("stock", 25)
            );
            database.getCollection("fruits").insertOne(
                    new Document("name", "Strawberry")
                            .append("price", 2.0)
                            .append("stock", 15)
            );
        }
    }
}

File: hw/chalan/u3/hw28PrinciplesAplication/FRUITS/JS/fruit-store/src/utils/MongoDBUtil.js
import { MongoClient } from 'mongodb'

class MongoDBUtil {
  static client
  static db

  static CONNECTION_STRING = 'mongodb+srv://kevin:kevin2001@cluster0.oxinj5p.mongodb.net/contac'
  static DATABASE_NAME = 'fruit_store'

  static async getDatabase() {
    if (!this.db) {
      this.client = new MongoClient(this.CONNECTION_STRING)
      await this.client.connect()
      this.db = this.client.db(this.DATABASE_NAME)
      await this.init()
    }
    return this.db
  }

  static async init() {
    const collections = await this.db.listCollections().toArray()
    if (!collections.find(c => c.name === 'fruits')) {
      await this.db.collection('fruits').insertMany([
        { name: 'Apple', price: 1.5, stock: 20 },
        { name: 'Banana', price: 0.8, stock: 30 },
        { name: 'Orange', price: 1.2, stock: 25 },
        { name: 'Strawberry', price: 2.0, stock: 15 }
      ])
    }
  }
}

export default MongoDBUtil
